domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapper.java

```diff
@@ -171,4 +179,41 @@ private AuthorisationToAccessControlListMapper() {
                 .collect(Collectors.toList());
         }
     }
+
+    // Mapper to call fieldTypeEntity listItems only for fixedList items not for all fields.
+    // It improves performance.
+    class ListItemsMapper {
+        private static String FIXED_LIST = "FixedList";
+        private ListItemsMapper() {
+            // Default constructor
+        }
+
+        static List<FixedListItem> map(FieldTypeEntity fieldTypeEntity) {
+            List<FixedListItem> fixedListItems = new ArrayList<>();
+
+            FieldTypeEntity baseFieldTypeEntity = fieldTypeEntity.getBaseFieldType();
+            if (baseFieldTypeEntity == null && FIXED_LIST.equalsIgnoreCase(fieldTypeEntity.getReference())) {
```

Comment:
`baseFieldTypeEntity == null` don't think it's possible for baseFieldTypeEntity to be null, so this might be not needed

Reply:
rather than:
```FIXED_LIST.equalsIgnoreCase(fieldTypeEntity.getReference())``` can you add an utility method to fieldTypeEntity called `isFixedList` please?